### PR TITLE
Operation should be resource/serializer independent

### PR DESCRIPTION
**Pull request by jozefvaclavik**
More context: https://public.3.basecamp.com/p/ZmTv4mzgTrpgy3thJW4zTKVn

TL;DR issues:
- passing serializer into operation ties it too close to single resource
- resource was not passed to operation, instead it has to be initialized and passed to serializer manually
- success expected (not required) serialized resource; its on dev if to serialize or no

TL;DR solution:
- we can determine domain that called operation (or pass domain into the operation)
- we can automatically pick up serializer for every resource (against that domain)
- only ActiveRecord objects should be serialized
- hashes and arrays are deep-serialized
- this way one can return multiple resources or primitive results instead of enforcing use of single serializer.

### CODE REVIEW

**Comment by KostasKostogloy** [comment]
@jozefvaclavik So based on your description, calling `success({ purchase: Purchase.last, reason: 'Testing' })` on the operation will return a hash with only purchase serialized and reason/string left untouched?

if that's the case, this is pretty sweet

**Comment by jozefvaclavik** [comment]
Yeah, thats correct. Only `ActiveRecord::Base` and `LedgerSync::Resource` subclasses gets serialized.
